cvmfs/catalog_mgr_rw.h

```diff
@@ -160,8 +174,72 @@ class WritableCatalogManager : public SimpleCatalogManager {
    * Directories don't have extended attributes at this point.
    */
   XattrList empty_xattrs;
+
+  /**
+   * Balancer thresholds
+   */
+  const bool is_balanced_;
+  const unsigned max_weight_;
+  const unsigned min_weight_;
+  const unsigned balance_weight_;
 };  // class WritableCatalogManager
 
+
+template <class CatalogMgrT>
```

Comment:
I think it's better to separate into to compilation units.  The `CatalogBalancer` has a right to get its own .cc and .h files.